src/Document/attributeParsers.ts

```diff
@@ -52,3 +54,20 @@ export function bzwVector3F(line: string): [number, number, number] {
     Number.parseFloat(tokens[2]),
   ];
 }
+
+export function bzwTeleRef(line: string): TeleporterReference {
+  let i = line.lastIndexOf(':'),
+    name = line.substring(0, i),
+    sideStr = line.substring(i + 1),
+    side =
+      sideStr == 'f'
+        ? TeleporterSide.Forward
+        : sideStr == 'b'
+        ? TeleporterSide.Backward
+        : TeleporterSide.Both;
```

Comment:
```suggestion
  const [name, sideStr] = line.split(':');
  let side = TeleporterSide.Both;
  
  if (sideStr === 'f') {
    side = TeleporterSide.Forward;
  } else if (sideStr === 'b') {
    side = TeleporterSide.Backward;
  }
```

You can use destructuring assignment instead of needing to do substrings. Also, a traditional if statement is easier to read than a nested ternary imo

Reply:
The reason I didn't do it that way is because the page here, https://wiki.bzflag.org/Link, shows an example where you can have a group name before a tele name, like "green:tele:*". So there's the potential to have at least two colons in the line.

Reply:
I personally don't mind nested ternaries, but I have no problem using if/else statements.

Reply:
> The reason I didn't use split() is because the page here, [wiki.bzflag.org/Link](https://wiki.bzflag.org/Link), shows an example where you can have a group name before a tele name, like "green:tele:*". So there's the potential to have at least two colons in the line.

Good catch on potentially having more than one `:`! Would you mind adding this unit test and taking into account that the `:*` is optional (according to bzfs behavior, not sure if the Wiki specifies)?

```ts
  it('should handle a link with no sides', () => {
    const bzwBody = `\
    link link0
      from tele1
      to tele2
    end
    `;

    const world = parseBZWDocument(bzwBody);
    const link: ITeleporterLink = Object.values(
      world.children,
    ).pop() as ITeleporterLink;

    expect(link.name).toEqual('link0');
    expect(link.from.name).toEqual('tele1');
    expect(link.from.side).toEqual(TeleporterSide.Both);
    expect(link.to.name).toEqual('tele1');
    expect(link.to.side).toEqual(TeleporterSide.Both);
  });
```